tests/rest/test_job.py

```diff
@@ -536,10 +541,28 @@ def job_with_1_asset(con100, requests_mock, tmp_path) -> BatchJob:
     requests_mock.get(API_URL + "/jobs/jj1/results", json={"assets": {
         "1.tiff": {"href": API_URL + "/dl/jjr1.tiff", "type": "image/tiff; application=geotiff"},
     }})
+    requests_mock.head(API_URL + "/dl/jjr1.tiff", headers={"Content-Length": f"{len(TIFF_CONTENT)}"})
     requests_mock.get(API_URL + "/dl/jjr1.tiff", content=TIFF_CONTENT)
+
     job = BatchJob("jj1", connection=con100)
     return job
 
+@pytest.fixture
+def job_with_chunked_asset_using_head(con100, requests_mock, tmp_path) -> BatchJob:
+    requests_mock.get(API_URL + "/jobs/jj1/results", json={"assets": {
+        "1.tiff": {"href": API_URL + "/dl/jjr1.tiff", "type": "image/tiff; application=geotiff"},
+    }})
+    requests_mock.head(API_URL + "/dl/jjr1.tiff", headers={"Content-Length": f"{len(TIFF_CONTENT)}", "Accept-Ranges": "bytes"})
+
+    chunk_size = 1000
+    for r in range(0, len(TIFF_CONTENT), chunk_size):
+        from_bytes = r
+        to_bytes = min(r + chunk_size, len(TIFF_CONTENT)) - 1
+        requests_mock.get(API_URL + "/dl/jjr1.tiff", request_headers={"Range": f"bytes={from_bytes}-{to_bytes}"},
+                     response_list = [{"status_code": 500, "text": "Server error"},
+                                      {"status_code": 206, "content": TIFF_CONTENT[from_bytes:to_bytes+1]}])
```

Comment:
This for-loop of mocked request is mimics the actual implementation very closely. 
I think it's more robust to avoid this tight coupling by creating a single request mock, but with a dynamic handler where you inspect the received request headers as documented in https://requests-mock.readthedocs.io/en/latest/response.html#dynamic-response

In this test file we have already an example of that:
https://github.com/Open-EO/openeo-python-client/blob/4aba08a6a758385a01e7d4b00ba5f9a00daa2655/tests/rest/test_job.py#L794-L800